vpc/service/allocate.go

```diff
@@ -23,67 +22,98 @@ import (
 	"github.com/Netflix/titus-executor/api/netflix/titus"
 )
 
-// this is the maximum number of IP addresses you are allowed to assign to an interface in EC2.
-const maxAddressesPerInterface = 50
-
-// The dummy interface is used for IP addresses that are allocated
-const staticDummyInterfaceDescription = "titus-static-dummy"
+const (
+	// The dummy interface is used for IP addresses that are allocated
+	staticDummyInterfaceDescription = "titus-static-dummy"
+	staticReservationDescription    = "titus-static-address"
+)
 
 var (
 	errAllocationNotFound = status.Error(codes.NotFound, "Could not find allocation")
-	errConsistencyIssue   = status.Error(codes.Unknown, "Database state inconsistent, no IPs available on ENI")
 )
 
-func (vpcService *vpcService) getStaticInterface(ctx context.Context, tx *sql.Tx, session *ec2wrapper.EC2Session, subnetID string) (*ec2.NetworkInterface, error) {
-	ctx, span := trace.StartSpan(ctx, "getStaticInterface")
+func getDummyStaticInterface(ctx context.Context, session *ec2wrapper.EC2Session, subnetID string) (*ec2.NetworkInterface, error) {
+	ctx, span := trace.StartSpan(ctx, "getDummyStaticInterface")
 	defer span.End()
 
-	var eniID string
-	// We can't use the first address, maxAddressesPerInterface - 1
-	row := tx.QueryRowContext(ctx, "SELECT eni_id FROM static_enis WHERE subnet_id = $1 AND (SELECT count(*) FROM ip_addresses WHERE home_eni = static_enis.eni_id) < $2 FOR UPDATE LIMIT 1", subnetID, maxAddressesPerInterface-1)
-	err := row.Scan(&eniID)
-	if err != nil && err != sql.ErrNoRows {
-		err = errors.Wrap(err, "Cannot fetch ENI from static enis set")
+	createNetworkInterfaceInput := ec2.CreateNetworkInterfaceInput{
+		Description:                    aws.String(staticDummyInterfaceDescription),
+		SubnetId:                       aws.String(subnetID),
+		Ipv6AddressCount:               aws.Int64(1),
+		SecondaryPrivateIpAddressCount: aws.Int64(1),
```

Comment:
Why is this number 1?

Reply:
thats to have 1 IPv4 and 1 IPv6 address for the static allocation.

Reply:
AWS always assigns one IPv4 address.